lib/client.js

```diff
@@ -132,11 +138,24 @@ export default class Client {
         'User-Agent': 'intercom-node-client/2.0.0'
       }
     };
+    // Append methods arguments
+    args = Object.assign({}, defaultArgs, args);
+    // Arguements to finally use for request
+    const requestArgs = Object.assign({}, this.requestOpts);
+    // Don't just use Object.assign(requestArgs, args) here because we need to handle special cases like .headers
```

Comment:
It sounds like what we're aiming for here is a deep merge rather than a shallow merge like Object.assign does. I think it'd be good to move this logic out of this function into a generic deep merge function or we could add the `merge` function from [lodash](https://lodash.com/docs/4.17.11#merge). What do you think?

Reply:
Also we could just go with:

```
args = Object.assign({}, this.requestOpts, defaultArgs, args);
args.headers = Object.assign({}, this.requestOpts.headers, args.headers);
```

As its only headers that are specified in the defaultArgs anyway?

Reply:
Looks like `lodash.merge` is only 64K so I think it should be fine to add. This is a server side SDK so we don't need to worry too much about adding dependencies.

Think it'd be good to go with the deep merge so we can get any additional object options for free. I think we should give precedence to the args passed in though. We don't want to allow things like the `json` key or URL to changeable